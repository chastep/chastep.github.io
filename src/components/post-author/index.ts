export { PostAuthor } from "./post-author";
