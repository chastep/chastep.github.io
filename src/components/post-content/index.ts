export { PostContent } from "./post-content";
