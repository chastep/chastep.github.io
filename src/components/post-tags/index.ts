export { PostTags } from "./post-tags";