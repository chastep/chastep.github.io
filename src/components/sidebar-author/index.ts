export { SidebarAuthor } from "./sidebar-author";