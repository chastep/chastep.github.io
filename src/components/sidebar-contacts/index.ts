export { SidebarContacts } from "./sidebar-contacts";
