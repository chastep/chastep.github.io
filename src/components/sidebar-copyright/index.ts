export { SidebarCopyright } from "./sidebar-copyright";
