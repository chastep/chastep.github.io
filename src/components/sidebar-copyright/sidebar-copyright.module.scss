@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.sidebar-copyright {
  color: hsl(var(--color-muted-foreground));
  font-size: variables.$typography-small-font-size;
}
