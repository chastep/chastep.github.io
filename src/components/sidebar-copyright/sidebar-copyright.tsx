import React, { type FC } from "react";

import * as styles from "./sidebar-copyright.module.scss";

type SidebarCopyrightProps = {
  copyright: string;
};

const SidebarCopyright: FC<SidebarCopyrightProps> = ({ copyright }) => (
  <div className={styles.sidebarCopyright}>{copyright}</div>
);

export { SidebarCopyright };
