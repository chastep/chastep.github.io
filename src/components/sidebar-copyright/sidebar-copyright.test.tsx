import React from "react";
import { test, describe, expect } from "bun:test";

import * as mocks from "@/mocks";
import { SidebarCopyright } from "@/components/sidebar-copyright";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

describe("SidebarCopyright", () => {
  test("renders correctly", () => {
    const props = { copyright: mocks.siteMetadata.site.siteMetadata.copyright };
    const tree = createSnapshotsRenderer(<SidebarCopyright {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with empty string", () => {
    const props = { copyright: "" };
    const tree = createSnapshotsRenderer(<SidebarCopyright {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
