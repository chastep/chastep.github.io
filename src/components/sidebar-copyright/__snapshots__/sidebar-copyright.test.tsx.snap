// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarCopyright renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    "All rights reserved.",
  ],
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;

exports[`SidebarCopyright renders correctly with empty string 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": null,
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;
